commit ed47f9c077af47d088a889903eaa3139e059bcc4 (origin/scarthgap-dev)
Author: F. Aragon <francisco.aragon@advantech.de>
Date:   Thu Apr 17 18:45:03 2025 +0200

    add support for rom-5720 2G (imx8mq)

    MP: pass
    MQ: issues with suspend to RAM, external watchdog

    Changes from:
    - https://github.com/ADVANTECH-Corp/uboot-imx - imx8LBVD0075 / adv_v2023.04_6.1.22-2.0.0
         lpddr4_timing_rom5720a1_2G.c
         lpddr4_timing_b0.c
    - https://github.com/ADVANTECH-Corp/linux-imx.git - imx8LBVD0075
         imx8mq-rom5720-a1.dts as rom5721-db5901.dts
    - https://github.com/nxp-imx/uboot-imx - lf_v2024.04
         recipes-bsp/u-boot/u-boot-imx/rom5720-db5901/0001-minimal-differences.patch
            From changes in files:
            arch/arm/dts/imx8mq-evk.dts
            board/freescale/imx8mq_evk/Makefile
            include/configs/imx8mq_evk.h

Observation: ad hoc patch

Upstream-Status: Inappropriate [other]
Signed-off-by: F. Aragon <francisco.aragon@advantech.de>

diff --git a/arch/arm/dts/imx8mq-evk.dts b/arch/arm/dts/imx8mq-evk.dts
index 632a83b5d0e..0607e489ad5 100644
--- a/arch/arm/dts/imx8mq-evk.dts
+++ b/arch/arm/dts/imx8mq-evk.dts
@@ -27,7 +27,7 @@
 
 	memory@40000000 {
 		device_type = "memory";
-		reg = <0x00000000 0x40000000 0 0xc0000000>;
+		reg = <0x00000000 0x40000000 0 0x80000000>;
 	};
 
 	pcie0_refclk: pcie0-refclk {
diff --git a/board/freescale/imx8mq_evk/Makefile b/board/freescale/imx8mq_evk/Makefile
index fd260bb1ec4..81d2d0ade76 100644
--- a/board/freescale/imx8mq_evk/Makefile
+++ b/board/freescale/imx8mq_evk/Makefile
@@ -8,9 +8,6 @@ obj-y += imx8mq_evk.o
 
 ifdef CONFIG_SPL_BUILD
 obj-y += spl.o
-ifdef CONFIG_IMX8MQ_4GB_DDR_TIMING
-obj-$(CONFIG_IMX8M_LPDDR4) += lpddr4_timing_4g.o
-else
-obj-$(CONFIG_IMX8M_LPDDR4) += lpddr4_timing_b0.o lpddr4_timing.o
-endif
+obj-$(CONFIG_IMX8M_LPDDR4) += lpddr4_timing_rom5720a1_2G.o
+obj-$(CONFIG_IMX8M_LPDDR4) += lpddr4_timing_b0.o
 endif
diff --git a/include/configs/imx8mq_evk.h b/include/configs/imx8mq_evk.h
index 562d56ee67a..31890d76811 100644
--- a/include/configs/imx8mq_evk.h
+++ b/include/configs/imx8mq_evk.h
@@ -147,9 +147,7 @@
 
 #define CFG_SYS_SDRAM_BASE           0x40000000
 #define PHYS_SDRAM                      0x40000000
-#define PHYS_SDRAM_SIZE			0xC0000000 /* 3GB DDR */
-#define PHYS_SDRAM_2			0x100000000
-#define PHYS_SDRAM_2_SIZE		0x40000000 /* 1GB */
+#define PHYS_SDRAM_SIZE			0x80000000 /* 2GB DDR */
 
 #define CFG_MXC_UART_BASE		UART_BASE_ADDR(1)
 
